src/components/TaskStatusBadge/index.tsx

```diff
@@ -0,0 +1,108 @@
+import { css } from "@emotion/react";
+import styled from "@emotion/styled";
+import Badge, { Variant } from "@leafygreen-ui/badge";
+import { palette } from "@leafygreen-ui/palette";
+import { taskStatusToCopy } from "constants/task";
+import { TaskStatus } from "types/task";
+import { reportError } from "utils/errorReporting";
+
+const { purple } = palette;
+
+interface BadgeColorProps {
+  border?: string;
+  fill?: string;
+  text?: string;
+}
+
+const badgeWidthMaxContent = css`
+  width: max-content;
+`;
+
+// only use for statuses whose color is not supported by leafygreen badge variants, i.e. SystemFailed, TestTimedOut, SetupFailed
+const StyledBadge = styled(Badge)<BadgeColorProps>`
+  ${({ border }) => border && `border-color: ${border} !important;`}
+  ${({ fill }) => fill && `background-color: ${fill} !important;`}
+  ${({ text }) => text && `color: ${text} !important;`}
+  ${badgeWidthMaxContent}
+`;
+
+interface TaskStatusBadgeProps {
+  status: string;
+}
+const TaskStatusBadge: React.VFC<TaskStatusBadgeProps> = ({ status }) => {
+  if (!status) {
+    return null;
+  }
+
+  let displayStatus = status;
+  if (taskStatusToCopy[status] === undefined) {
+    const err = new Error(`Status '${status}' is not a valid task status`);
+    reportError(err).warning();
+  } else {
+    displayStatus = taskStatusToCopy[status];
+  }
```

Comment:
Kind of wondering if we need this, though I know it's directly from Spruce

```suggestion
  const displayStatus = taskStatusToCopy[status] ?? status;
```

Reply:
Since we're reporting the error in Spruce it seems reasonable to me to omit this logic in Parsley